plugins/async-node/ios/Tests/AsynNodePluginTests.swift

```diff
@@ -292,6 +284,166 @@ class AsyncNodePluginTests: XCTestCase {
         XCTAssert(count == 3)
         XCTAssertEqual(expectedMultiNode2Text, "new node from the hook 2")
     }
+
+    func testAsyncNodeWithNullNode() {
+        let handlerExpectation = XCTestExpectation(description: "null node handled")
+
+        let context = JSContext()
+
+        let resolveHandler: AsyncHookHandler = { node in
+            handlerExpectation.fulfill()
+            XCTAssertNil(node)
+            return .singleNode(.concrete(JSValue()))
+        }
+
+        let asyncNodePluginPlugin = AsyncNodePluginPlugin()
+        let plugin = AsyncNodePlugin(plugins: [asyncNodePluginPlugin], resolveHandler)
+
+        plugin.context = context
+
+        XCTAssertNotNil(asyncNodePluginPlugin.context)
+
+        let player = HeadlessPlayerImpl(plugins: [ReferenceAssetsPlugin(), plugin], context: context ?? JSContext())
+
+        player.start(flow: .asyncNodeJson, completion: {_ in})
+
+        wait(for: [handlerExpectation], timeout: 5)
```

Comment:
Where is the replacement value tested? We'd want to have similar assertions to the other platform tests, where we test the resolution did/or did not trigger another view update and what the view tree looks like after the update happens.